Aptoide/aptoide-client-v8
# [Feature/ASV-994] Reactions custom views + interactor/network model

**What does this PR do?**

   This adds the possibility of showing reactions inside vanilla. They are currently not being shown anywhere. 

   You should look for 2 different parts in this PR. The custom views (entry point is ReactionsPopup.java) and the reactions model (entry point being the Reactions.java).

   The reactions custom views are done by having a custom linearlayout (ReactionsView) that has 5 reaction objects in it. Each reaction object is rendered by lottie using LottieAnimationView and added to ReactionsView. Appropriate callbacks were done in order to return information about what reaction was clicked. Check ReactionsPopup Documentation to see if it is clear.

   2nd phase was the Reactions interactor that at the moment just has an interface to set a reaction with an id and a reactiontype (might be subject of change in the future). The SetReactionsRequest loadDataFromNetwork is throwing an exception on purpose because there is no webservice available yet to do this. Thats also the reason why there is a ReactionFakeService - so we can continue the implementation without the web dependency.

  ReactionsPopup show() currently only supports api19+. This will need to be handled when we show reactions in the future. 

**Database changed?**

   No

**Where should the reviewer start?**

- [ ] ReactionsPopup.java
- [ ] Reactions.java

**How should this be manually tested?**

   To check the reactions custom view, just show it in any button of the application. For instance, I've tested it in the rate button in the appview and then shown the reactions view.

**What are the relevant tickets?**

  Tickets related to this pull-request: [ASV-994](https://aptoide.atlassian.net/browse/ASV-994)


**Code Review Checklist**
